test_specs.sh

```
@@ -0,0 +1,102 @@
+# Assign arguments to variables
+source_dir="${1%/}"
+specs_dir="${2%/}"
+output_dir="${3%/}"
+
+# Function to process files
+process_files() {
+    local src_dir="$1"
+    local out_dir="$2"
+
+    find "$src_dir" -type f -name "*.gv.txt" -print0 | while IFS= read -r -d $'\0' file; do
+        rel_path="${file#$src_dir/}"
```

Shellcheck

[info] Expansions inside ${..} need to be quoted separately, otherwise they match as patterns.

(SC2295)